File: GrammerBasics/src/Preposition.java

public class Preposition extends Grammer {

	String prep;
	NounPhrase phrase; 
	public Preposition() {
		
	}
	public Preposition(String prep) {
		this.prep = prep;
	}
	public NounPhrase getPhrase() {
		return phrase;
	}
	public void setPhrase(NounPhrase phrase) {
		this.phrase = phrase;
	}
	public String getPrep() {
		return prep;
	}
	public void setPrep(String prep) {
		this.prep = prep;
	}
	@Override
	public String printSelf() {
		
		return prep;
	}
	@Override
	public String toString() {
		return prep + " "+ phrase.toString();
	}
	

}

File: JabberJaw/Grammer/Preposition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Grammer
{
    class Preposition : Grammer
    {

        String prep;
        NounPhrase phrase;

    public NounPhrase getPhrase()
    {
        return phrase;
    }
    public void setPhrase(NounPhrase phrase)
    {
        this.phrase = phrase;
    }
    public String getPrep()
    {
        return prep;
    }
    public void setPrep(String prep)
    {
        this.prep = prep;
    }

        public override string printSelf()
        {
            return prep;
        }
    }

}
